python/cuml/dask/neighbors/nearest_neighbors.py

```diff
@@ -187,8 +192,7 @@ def _query_models(self, n_neighbors,
                         workers=[worker]))
                        for idx, worker in enumerate(comms.worker_addresses)])
 
-        wait(list(nn_fit.values()))
-        raise_exception_from_futures(list(nn_fit.values()))
+        wait_and_raise_from_futures(list(nn_fit.values()))
```

Comment:
I guess we still have the general question - should all model fits wait before returning? Or is that the caller's job so they could overlap things if they want?

Reply:
Unfortunately, they do need to wait so that the comms can be closed after the fit is called.

Reply:
You’ve got me thinking about this one now. One other challenge here is that all the futures will disappear as soon as the function is returned. I do like the general idea, though. I’ve created https://github.com/rapidsai/cuml/issues/2416 to ponder and discuss it further.